model/src/main/kotlin/io/spine/validation/bound/RangeOption.kt

```diff
@@ -109,18 +109,17 @@ internal class RangePolicy : Policy<FieldOptionDiscovered>() {
         val fieldType = checkFieldType(field, file, RANGE)
 
         val option = event.option.unpack<RangeOption>()
-        val (messageType, _) = typeSystem.findMessage(field.declaringType)!!
-        val context = RangeContext(option.value, typeSystem, messageType, fieldType, field, file)
-        val delimiter = context.checkDelimiter()
+        val optionMetadata = RangeOptionMetadata(option.value, field, fieldType, file, typeSystem)
+        val delimiter = optionMetadata.checkDelimiter()
```

Comment:
Why do we need two steps for splitting the range? We don't seem to use the `delimiter` outside the calls to `optionMetadata`. 

Can we encapsulate it somehow, please?

Reply:
It is better to have these explanations "embedded" in the code, rather than left as a comment to a PR. I'll try to explain it differently.

The code like this `left.substring(1)` looks mysterious. We cut the first character in something `left`. Why? For what? It is better to have a variable which explains the result of the operation. It would help understanding the code of parsing, which is arguably one of the complicated things we do in this project.